activerecord/lib/active_record/enum.rb

```diff
@@ -158,7 +159,14 @@ def assert_valid_value(value)
         attr_reader :name, :mapping
     end
 
-    def enum(definitions)
+    def enum(name = nil, values = nil, **options)
+      if name
+        values ||= options.slice!(:prefix, :suffix, :scopes, :default)
+        return _enum(name, values, **options)
+      end
+
+      definitions = options
+
       prefix = definitions.delete(:_prefix)
```

Comment:
Should we deprecate all those options and the old syntax? We can probably include a Cop to do the migration automatically.

Reply:
It is a good idea to me if happening the migration is ok for us.

I thought since it is confusing that the old syntax options has leading `_` but the new syntax options doesn't have that, I'd also like to deprecate defining  enum on the attribute names (`:prefix`, `:suffix`, `:scopes`, and `:default`) in the old syntax to reserve the names to make it as options in the future.

Reply:
We don't need to deprecate in this release, but if we can easily autocorrect the deprecation with a cop, I think we should.